src/AdvancedPS.jl

```diff
@@ -0,0 +1,66 @@
+### Attention this is a development package! It wount run.
+
+
+module AdvancedPS
+
+
+
+        const DEBUG = Bool(parse(Int, get(ENV, "DEBUG_APS", "0")))
+        using Libtask
+        using StatsFuns: logsumexp, softmax!
+        import Base.copy
+
+        abstract type AbstractTaskInfo end
+        abstract type AbstractParticleContainer end
+        abstract type AbstractTrace end
+
+
+        include("tasks.jl")
+
+        export  ParticleContainer,
+                Trace,
+                weights,
+                logZ,
+                current_trace,
+                extend!,
+                empty!
+
+
+        include("resample.jl")
+
+        export resample!
+
+
+
+
+        include("taskinfo.jl")
+
+        export  PGTaskInfo,
+                PGASTaskInfo
+
+        include("samplers.jl")
+
+        export sampleSMC!, samplePG!
+
+
+        include("resample_functions.jl")
+
+        export  resample,
+                randcat,
+                resample_multinomial,
+                resample_residual,
+                resample_stratified,
+                resample_systematic
+
+        include("Interface.jl")
+
+        export  report_transition,
+                report_observation,
+                init,
```

Comment:
`init` seems a bit too generic to be exported. Maybe rename it?

Reply:
Good point, I will most probably remove the whole Interface.